### PR TITLE
CMake: Fix configuration on Win32

### PR DESCRIPTION
**Pull request by myd7349**
On Windows with MSVC, this line:
```cmake
check_library_exists(m pow "" WITH_HDRHISTOGRAM)
```
causes `WITH_HDRHISTOGRAM` set to `OFF`.
As a result:
- When specifying `WITHOUT_WIN32_CONFIG=ON`, the histogram source will not be built.
- When specifying `WITHOUT_WIN32_CONFIG=OFF`, there will be link erros since `WITH_HISTOGRAM` is defined as 1 in `win32_config.h`

References:
https://stackoverflow.com/questions/32816646/can-cmake-detect-if-i-need-to-link-to-libm-when-using-pow-in-c

### CODE REVIEW

**Comment by myd7349** [comment]
The configuration step on Win32 through CMake still have room for improvement.
1.

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,7 +35,12 @@ endif()
 
 # LIBM {
 include(CheckLibraryExists)
-check_library_exists(m pow "" WITH_HDRHISTOGRAM)
+check_library_exists(m pow "" HAVE_LIBM)
```
Comment (edenhill) [inline CMakeLists.txt]: On MSVC we don't need to, and probably shouldnt, check for libm at all.
What do you think?

**Comment by edenhill** [comment]
I think I like the more aggressive approach, better to get this all right in one go.

File: CMakeLists.txt (hunk 1 above)
Comment (myd7349) [reply CMakeLists.txt]: @edenhill Thanks for your review! I will update this PR later.

**Comment by myd7349** [comment]
Here is a related discussion about `libm`: https://github.com/mesonbuild/meson/issues/5390#issuecomment-491640529

**Comment by myd7349** [comment]
`snappy` support doesn't work out-of-box, reported at:
https://github.com/microsoft/vcpkg/pull/7456

**Comment by edenhill** [comment]
@myd7349 WITH_SNAPPY should always be set to 1 in CMakeLists.txt

**Pull request closed without merging**